docs/tutorials/streaming_analytics.md

```diff
@@ -0,0 +1,460 @@
+will contain exactly one reading -- the pre-aggregated maximum computed by `MyReducingMax`.
+
+### Late Events
+
+By default, when using event time windows, late events are dropped. There are two optional parts of
+the window API that give you more control over this.
+
+You can arrange for the events that would be dropped to be collected to an alternate output stream
+instead, using a mechanism called [Side Outputs]({{ site.baseurl }}{% link tutorials/event_driven.md
+%}#side-outputs). Here's an example of what that might look like:
+
+{% highlight java %}
+OutputTag<Event> lateTag = new OutputTag<Event>("late"){};
+
+SingleOutputStreamOperator<Event> result = stream.
+    .keyBy(...)
+    .window(...)
+    .sideOutputLateData(lateTag)
+    .process(...);
+  
+DataStream<Event> lateStream = result.getSideOutput(lateTag);
+{% endhighlight %}
+
+You can also specify an interval of _allowed lateness_ during which the late events will continue to
+be assigned to the appropriate window(s) (whose state will have been retained). By default each late
+event will cause a late firing of the window function.
+
+By default the allowed lateness is 0. In other words, elements behind the watermark are dropped (or
+sent to the side output).
+
+For example:
+
+{% highlight java %}
+stream.
+    .keyBy(...)
+    .window(...)
+    .allowedLateness(Time.seconds(10))
+    .process(...);
+{% endhighlight %}
+
+When the allowed lateness is greater than zero, only those events that are so late that they would
+be dropped are sent to the side output (if it has been configured).
+
+### Surprises
+
+Some aspects of Flink's windowing API may not behave in the way you would expect. Based on
+frequently asked questions on the [flink-user mailing
+list](https://flink.apache.org/community.html#mailing-lists) and elsewhere, here are some facts
+about windows that may surprise you.
+
+#### Sliding Windows Make Copies
+
+Sliding window assigners can create lots of window objects, and will copy each event into every
+relevant window. For example, if you have sliding windows every 15 minutes that are 24-hours in
+length, each event will be copied into 4 * 24 = 96 windows.
+
+#### Time Windows are Aligned to the Epoch
+
+Just because you are using hour-long processing-time windows and start your application running at
+12:05 does not mean that the first window will close at 1:05. The first window will be 55 minutes
+long and close at 1:00.
+
+#### Windows Can Follow Windows
+
+For example, it works to do this:
+
+{% highlight java %}
+stream
+    .keyBy(t -> t.key)
+    .timeWindow(<time specification>)
+    .reduce(<reduce function>)
+    .timeWindowAll(<same time specification>)
+    .reduce(<same reduce function>)
+{% endhighlight %}
+
+You might expect Flink's runtime to be smart enough to do this parallel pre-aggregation for you
+(provided you are using a ReduceFunction or AggregateFunction), but it's not.
```

Comment:
Add requirements on the time specifications to be compatible?

Reply:
done